arch/arm/cp15.c

```diff
@@ -1,45 +1,53 @@
 #include <stdio.h>
 #include <arch/armv7.h>
-
-#define __CP32(cp, n, opc1, m, opc2)    (n << 12 | opc1 << 8 | m << 4 | opc2)
-#define CP32(args...)                   __CP32(args)
+#include <drivers/vdev/vdev_timer.h>
 
 #define opc2_bit                (0x7 << 17)
 #define opc1_bit                (0x7 << 14)
 #define crn_bit                 (0xf << 10)
-#define crm_bit                 (0xf << 1)
-
-#define opc2(iss)               ((iss & opc2_bit) >> 17)
-#define opc1(iss)               ((iss & opc1_bit) >> 14)
-#define crn(iss)                ((iss & crn_bit)  >> 10)
-#define crm(iss)                ((iss & crm_bit)  >> 1)
+#define rt_bit                  (0xf <<  5)
+#define crm_bit                 (0xf <<  1)
 
-#define decode_cp15(iss)        crn(iss) << 12 | opc1(iss) << 8 | \
-                                crm(iss) << 4  | opc2(iss)
+#define opc2(iss)               (((iss) & opc2_bit) >> 17)
+#define opc1(iss)               (((iss) & opc1_bit) >> 14)
+#define crn(iss)                (((iss) & crn_bit)  >> 10)
+#define rt(iss)                 (((iss) & rt_bit)   >>  5)
+#define crm(iss)                (((iss) & crm_bit)  >>  1)
 
 #define decode_dir(iss)         (iss & 1)
-#define decode_rt(iss)          ((iss & 0x1E0) >> 5)
+#define decode_cp15(iss)        __CP32(15, crn(iss), opc1(iss), crm(iss), opc2(iss))
```

Comment:
This macro was originally written by you and I just modified it a bit.

And command scheme seems similar but not the same, so we need separate definitions for other co-processor accesses.

So I'd say let's leave it as is.